lib/pavilion/test_run.py

```diff
@@ -923,6 +929,80 @@ def create_id_dir(id_dir):
     def __repr__(self):
         return "TestRun({s.name}-{s.id})".format(s=self)
 
+    def _get_skipped(self):
+        match_list = []
+        match_list = self._match(match_list)
+
+        if len(match_list) is 0:
+            return False
+        else:
+            self.status.set(STATES.COMPLETE, match_list[0])
+            return True
+
+    def _make_regex(self, cond):
+        """_make_regex takes a dictionary and checks each value
+        under all keys and appends/prepends [^$] to make sure all
+        values are compatible with the regex match for conditional
+        statements. This is needed if the user is not using regex.
+        It explicitly defines the given value and only matches to it.
+        :param dict: Dictionary containing conditional statements
+        :return dict: Dictionary with updated values."""
+        cond_dict = self.config.get(cond, {})
+        for key in cond_dict:
+            list = []
+            for value in cond_dict[key]:
+                if len(value) is 0:
+                    value = '^$'
+                if value[0] is '^':
+                    pass
+                else:
+                    value = ''.join(['^', value])
+                if value[len(value)-1] is '$':
+                    pass
+                else:
+                    value = ''.join([value, '$'])
```

Comment:
Since you're dealing with strings though, there's a much cleaner way to do this whole thing:
```python
if not value.startswith('^'):
  value = '^' + value
if not value.endswith('$'):
  value = value + '$'
```

Reply:
That make this whole process so clean and short, I think it can be inserted straight into your loops below.

Reply:
in fact, when you more carefully read the regular expression documentation for 'match' and 'search', you'll see only the second check is necessary.